service/history/shardController.go

```diff
@@ -322,25 +322,38 @@ func (c *shardController) acquireShards() {
 	sw := c.metricsClient.StartTimer(metrics.HistoryShardControllerScope, metrics.AcquireShardsLatency)
 	defer sw.Stop()
 
-AcquireLoop:
+	concurrency := common.MaxInt(c.config.AcquireShardConcurrency(), 1)
+	shardActionCh := make(chan int, c.config.NumberOfShards)
```

Comment:
I think you don't need a channel of length NumberOfShards if you move the loop that enqueues the shardIDs into the shardActionCh after the loop that spins up goroutines. The channel could be any size > 0 but channel of size=concurrency might be optimal.

Reply:
Done